test/e2e/egress_test.go

```diff
@@ -716,6 +736,59 @@ func hasIP(data *TestData, nodeName string, ip string) (bool, error) {
 	return strings.Contains(stdout, ip+"/32") || strings.Contains(stdout, ip+"/128"), nil
 }
 
+func setupIPNeighborChecker(data *TestData, t *testing.T, observerNode, node1, node2, ip string) (checkIPNeighbor func(string), cleanup func(), err error) {
+	transportInterface, err := data.GetTransportInterface()
+	require.NoError(t, err)
+
+	macAddress1, err := data.GetNodeMACAddress(node1, transportInterface)
+	require.NoError(t, err)
+	macAddress2, err := data.GetNodeMACAddress(node2, transportInterface)
+	require.NoError(t, err)
+	nodeToMACAddress := map[string]string{node1: macAddress1, node2: macAddress2}
+
+	antreaPodName, err := data.getAntreaPodOnNode(observerNode)
+	require.NoError(t, err)
+
+	cmd := []string{"ip", "route", "replace", ip, "dev", transportInterface}
+	_, _, err = data.RunCommandFromPod(antreaNamespace, antreaPodName, agentContainerName, cmd)
+	require.NoError(t, err, "Failed to add a direct route for Egress IP %s on Node %s", ip, observerNode)
```

Comment:
Added the comment:
```
// The Egress IP may not be in the same subnet as the primary IP of the transport interface.
// Adding a direct route for the Egress IP so the Node will query its MAC address, instead of trying to connect via
// its gateway.
```

Reply:
> The Egress IP may not be in the same subnet as the primary IP of the transport interface.

So we are getting ARP messages (which means that the observer Node is in the same L2 network as the Egress Node's transport interface), but the path from the observer Node to the Egress IP goes through a gateway? That seems strange, do we have that configuration in one of our testbeds?

Reply:
I guess that makes sense if we have 2 separate subnets for transport IPs and Egress IPs. And the gateway is configured correctly with both subnets, but the Nodes are not ware of the Egress subnet,

Reply:
Yes, we use a reserved IP subnet for Egress IPs, which is to prevent IP conflict with other network entities in the test network, and to make it simpler when allocating Egress IPs given the testbeds' IPs are quite random.